eclipse/src/saros/ui/wizards/AddProjectToSessionWizard.java

```diff
@@ -668,4 +687,106 @@ private void displaySaveDirtyEditorsDialog(final List<IEditorPart> dirtyEditors)
       for (IEditorPart editor : dirtyEditors) editor.doSave(new NullProgressMonitor());
     }
   }
+
+  private void storeCurrentProjectNameMapping(final JID jid) {
+    Map<String, String> currentProjectNameMapping = new HashMap<>();
+
+    for (final ProjectNegotiationData data : negotiation.getProjectNegotiationData()) {
+      final String projectID = data.getProjectID();
+      currentProjectNameMapping.put(
+          data.getProjectName(), namePage.getTargetProjectName(projectID));
+    }
+
+    mappingStorage.updateMapping(jid, currentProjectNameMapping);
+  }
+
+  private static final class ResourceMappingStorage {
+    private static final String RESOURCE_MAPPING_STORE_PREFIX = "resource.mapping.storage.";
+
+    private static final CharSequence DELIMITER = ":";
+
+    private static final CharSequence ESCAPE_CHARACTER = "\\";
+
+    private static final CharSequence ESCAPE_SEQUENCE =
+        ESCAPE_CHARACTER.toString() + DELIMITER.toString();
+
+    private static final Pattern SPLIT_PATTERN =
+        Pattern.compile(
+            "(?<!"
+                + Pattern.quote(ESCAPE_CHARACTER.toString())
+                + ")"
+                + Pattern.quote(DELIMITER.toString()));
+
+    private final IPreferenceStore store;
+
+    public ResourceMappingStorage(final IPreferenceStore store) {
+      this.store = store;
+    }
+
+    public Map<String, String> getMapping(final JID jid) {
+      final Map<String, String> result = new HashMap<>();
+
+      final String mapping = store.getString(RESOURCE_MAPPING_STORE_PREFIX + jid.getBase());
+
+      if (mapping.isEmpty()) return result;
+
+      final String[] names = SPLIT_PATTERN.split(mapping);
```

Comment:
Check whether array length is even to avoid index out of bound exception in case of data corruption? But I am not sure whether this is really necessary.

Reply:
Should be 

for (int i = 0; i < names.length -1; i +=2)

instead of 

for (int i = 0; i < names.length; i +=2)